Modules/FT0/src/DigitQcTask.cxx

```diff
@@ -310,78 +305,57 @@ void DigitQcTask::startOfCycle()
 
 void DigitQcTask::monitorData(o2::framework::ProcessingContext& ctx)
 {
-  double curTfTimeMin = -1;
-  double curTfTimeMax = 0;
   mTfCounter++;
   auto channels = ctx.inputs().get<gsl::span<o2::ft0::ChannelData>>("channels");
   auto digits = ctx.inputs().get<gsl::span<o2::ft0::Digit>>("digits");
-
+  if (digits.size() > 0) {
+    // Considering that Digit container is already sorted by IR
+    const o2::InteractionRecord& firstIR = digits[0].getIntRecord();
+    const o2::InteractionRecord& lastIR = digits[digits.size() - 1].getIntRecord();
+    auto timeMinNS = firstIR.bc2ns();
+    auto timeMaxNS = firstIR.bc2ns();
+    mTimeMinNS = std::min(mTimeMinNS, timeMinNS); // to be sure if somehow TFID is unordered
+    mTimeMaxNS = std::max(mTimeMaxNS, timeMaxNS);
+    mTimeSum += timeMaxNS - timeMinNS;
+  }
   for (auto& digit : digits) {
     const auto& vecChData = digit.getBunchChannelData(channels);
     bool isTCM = true;
-    mTimeCurNS = o2::InteractionRecord::bc2ns(digit.getBC(), digit.getOrbit());
-    if (mTimeMinNS < 0)
-      mTimeMinNS = mTimeCurNS;
-    if (mTimeCurNS < mTimeMinNS)
-      mTimeMinNS = mTimeCurNS;
-    if (mTimeCurNS > mTimeMaxNS)
-      mTimeMaxNS = mTimeCurNS;
-
-    if (curTfTimeMin < 0)
-      curTfTimeMin = mTimeCurNS;
-    if (mTimeCurNS < curTfTimeMin)
-      curTfTimeMin = mTimeCurNS;
-    if (mTimeCurNS > curTfTimeMax)
-      curTfTimeMax = mTimeCurNS;
-
-    if (digit.mTriggers.amplA == -5000 && digit.mTriggers.amplC == -5000 && digit.mTriggers.timeA == -5000 && digit.mTriggers.timeC == -5000)
+    if (digit.mTriggers.amplA == -5000 /*&& digit.mTriggers.amplC == -5000 && digit.mTriggers.timeA == -5000 && digit.mTriggers.timeC == -5000*/) {
```

Comment:
Hi Artur,
Do you know where the value of -5000 was taken from?
The default amplitude (according to https://github.com/AliceO2Group/AliceO2/blob/dev/DataFormats/Detectors/FIT/common/include/DataFormatsFIT/Triggers.h#L45 ) is 0, not -5000. It used to be -5000 though, so it may be a leftover. Have you tested that this condition works as expected?

Reply:
Hi @mslupeck , ah, yes, you are right, I took condition from previous QC version. So in previous version we had the same situation. Ok, lets keep this in current PR(due to CICD is already passed). I'll prepare hotfix.